src/osemgrep/reporting/Cli_json_output.ml

```diff
@@ -390,6 +443,40 @@ let dedup_and_sort (xs : Out.cli_match list) : Out.cli_match list =
            Hashtbl.replace seen key true;
            true)
 
+(* This is the same algorithm for indexing as in cli. We shouldn't need to update this *)
```

Comment:
what is cli in this context? Do you mean pysemgrep?

Reply:
Can you add some comments explaning why we need this function? And what it does?